baiji/copy.py

```diff
@@ -51,19 +51,35 @@ def etag(self):
             return self.connection.etag(self.uri)
         def rm(self):
             return self.connection.rm(self.uri)
-        def lookup(self):
+        def lookup(self, version_id=None):
+            from boto.exception import S3ResponseError
+            from baiji.exceptions import InvalidVersionID
+
             if self.is_file:
                 raise ValueError("S3CopyOperation.CopyableKey.lookup called for local file")
-            key = self.bucket.lookup(self.remote_path)
+
+            key = None
+
+            try:
```

Comment:
This is functionality that I expect we'll wind up needing in many places as we add support for versions; probably worth wrapping it in a utility function.